function-app-deploy/entrypoint.sh

```diff
@@ -2,80 +2,90 @@
 
 set -e
 
-LINUX_APP_SUBSTRING='linux'
-
-if [[ -z $WEB_APP_NAME ]];
+if [[ -z $AZURE_APP_NAME ]];
 then 
-    echo "Required Web App Name. Provide value in WEB_APP_NAME variable" >&2
+    echo "Required Azure Function App name. Provide value in AZURE_APP_NAME variable" >&2
+    exit 1
+fi
+
+echo "Azure Function App Name: ${AZURE_APP_NAME}"
+
+if [[ -z $AZURE_APP_PACKAGE_LOCATION ]];
+then
+    echo "Package location required. Provide value in AZURE_APP_PACKAGE_LOCATION variable" >&2
+    exit 1
+fi
+
+if [[ ! -d "$AZURE_APP_PACKAGE_LOCATION" &&  ! -f "$AZURE_APP_PACKAGE_LOCATION" ]];
+then
+    echo "Package location '$AZURE_APP_PACKAGE_LOCATION' does not exist."
     exit 1
 fi
-echo "Web App Name: ${WEB_APP_NAME}"
 
-RESOURCE_GROUP_NAME=`az resource list -n "${WEB_APP_NAME}" --resource-type "Microsoft.Web/Sites" --query '[0].resourceGroup' | xargs`
+RESOURCE_GROUP_NAME=`az resource list -n "${AZURE_APP_NAME}" --resource-type "Microsoft.Web/Sites" --query '[0].resourceGroup' | xargs`
+
 if [[ -z $RESOURCE_GROUP_NAME ]];
 then
-    echo "Web App '${WEB_APP_NAME}' should exist before deployment." >&2
+    echo "Web App '${AZURE_APP_NAME}' should exist before deployment." >&2
     exit 1
 fi
 
 echo "Resource Group Name: ${RESOURCE_GROUP_NAME}"
 
-APP_KIND=`az resource show -n "${WEB_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" --resource-type "Microsoft.Web/Sites" --query 'kind'` 
+echo "Provided package path: '${AZURE_APP_PACKAGE_LOCATION}'"
+
+if [[ -d $AZURE_APP_PACKAGE_LOCATION ]];
+then
+    NEW_PACKAGE_LOCATION="${GITHUB_WORKSPACE}/package_$RANDOM.zip"
+    echo "Compressing Package '${AZURE_APP_PACKAGE_LOCATION}' to '$NEW_PACKAGE_LOCATION'"
+    cd $AZURE_APP_PACKAGE_LOCATION
+    zip -r "$NEW_PACKAGE_LOCATION" * > /dev/null
+    AZURE_APP_PACKAGE_LOCATION="$NEW_PACKAGE_LOCATION"
+    cd "$GITHUB_WORKSPACE"
+    echo "Compressed package. New Package path: '${AZURE_APP_PACKAGE_LOCATION}'"
+fi
+
+APP_KIND=`az resource show -n "${AZURE_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" --resource-type "Microsoft.Web/Sites" --query 'kind'` 
+
 echo "Web App type : ${APP_KIND}"
 
 echo "Initiating Web App Deployment"
+
 if [[ ! $APP_KIND =~ $LINUX_APP_SUBSTRING ]];
```

Comment:
Variable not initialized LINUX_APP_SUBSTRING

Reply:
updated. thanks